src/dist/replication/meta_server/meta_server_failure_detector.cpp

```diff
@@ -263,6 +263,10 @@ void meta_server_failure_detector::on_ping(const fd::beacon_msg& beacon, ::dsn::
     {
         ack.is_master = false;
         ack.primary_node = get_primary();
+        {
```

Comment:
Then I think we should fix R instead of M. R should not consider itself alive when M is not the master. Right? (the default safety assumption should be: M considers R alive, while R considers itself dead even when they are not connected).